core/clients/key_flow.go

```diff
@@ -0,0 +1,359 @@
+package clients
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v2"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+const (
+	// Service Account Key Flow
+	// Auth flow env variables
+	ServiceAccountKey     = "STACKIT_SERVICE_ACCOUNT_KEY"
+	PrivateKey            = "STACKIT_PRIVATE_KEY"
+	ServiceAccountKeyPath = "STACKIT_SERVICE_ACCOUNT_KEY_PATH"
+	PrivateKeyPath        = "STACKIT_PRIVATE_KEY_PATH"
+)
+
+var tokenAPI = "https://api.stackit.cloud/service-account/token" //nolint:gosec // linter false positive
```

Comment:
What does the linter say?

Reply:
Also, is it ok, to have the link exposed like this?

Reply:
This is the rule that is triggered: https://securego.io/docs/rules/g101

We need to have this url by default for convenience of the user. I think there is no problem in exposing it, but I will double check